README.md

```diff
@@ -0,0 +1,265 @@
+doesn't have authorization, etc. Extensive mocking, table-driven tests, fuzzing,
+and mutation testing could be added later as unintentional error conditions are
+discovered, but are not within scope at this time.
+
+Since an adequate starting point for gRPC with mTLS is provided as an example in
+the
+[`grpc-go` repository,](https://github.com/grpc/grpc-go/tree/master/examples/features/encryption/mTLS)
+that will be used as an initial base for the server and client implementations
+to grow out of.
+
+## Usage Examples
+
+### Environment Variables
+
+#### Server
+```sh
+TP_WORKER_IP=127.0.0.1                          # --ip-addr
+TP_WORKER_PORT=8080                             # --port
+TP_WORKER_USER_CERTDIR=/path/to/cert/dir/       # --user-certs
+TP_WORKER_PRIVATE_KEY=/path/to/private/key.pem  # --private-key
+```
+
+#### Client
+```sh
+TP_WORKER_IP=127.0.0.1                                # --ip-addr
+TP_WORKER_PORT=8080                                   # --port
+TP_WORKER_CLIENT_PRIVATE_KEY=/path/to/private/key.pem # --private-key
+```
+
+### Start the Server
+
+Starting the server attaches to server log output unless sent to the background.
+
+```sh
+tp-worker-server
+...
+< Logged Output >
+...
+```
+
+### Client Pings Server
+
+```sh
+tp-worker-client ping
+Server OK at 127.0.0.1:8080
+```
+
+### Client Starts a Job
+`ls -lha` used as example.
+
+The `--name` argument can be passed to provide a unique name to the job,
+otherwise we assign the job a unique name based on the command and an increment.
+Names must start with a letter to prevent confusion with pid's elsewhere. Names
+are global. Fails if the name is already taken.
+
+The `-f` or `--follow` argument is equivalent to `tail -f`, which immediately
+attaches to the streaming output of the job.
+
+```sh
+tp-worker-client start 'ls -lha'
+Server 127.0.0.1:8080: Started job 'ls1' at pid 12345: ls -lha
+```
+
+### Client Stops a Job
+
+Either a name or pid is required. Since a name must start with a letter we
+should be able to do this without requiring either `--name` or `--pid`
+arguments, though they could be available to make people feel better.
+
+The `-s` or `--signal` argument is equivalent to `kill -s`, where the user may
+select the signal to be sent to the process. Can accept either the signal
+name or number, such as `SIGKILL` or equivalently `9`. Defaults to `SIGTERM`
+or `15`. See `Signal numbering for standard signals` in `signal(7)` manpage
+for the full list of signals.
+
+```sh
+tp-worker-client stop ls1
+
+# or
+
+tp-worker-client stop 12345
+Server 127.0.0.1:8080: Stopped job 'ls1' at pid 12345: ls -lha
+```
+
+### Client Checks Job Status
+
+Possible statuses:
+* Queued (after queue tracking has been added to scope)
+* Started (the job has been sent to the system, but no output received yet)
+* Running (the job is running and output is being received)
+* Hung (the job is running, but no new output has been received for some time)
```

Comment:
No output doesn't necessarily mean hung. The job could be a webserver that's logging to a file instead of standard out, for example.